packages/vuetify/src/effects/elevation/index.ts

```diff
@@ -0,0 +1,33 @@
+// Setup
+import { computed } from 'vue'
+
+// Types
+interface ElevationProps {
+  elevation?: number
+}
+
+// Props
+export const ElevationProps = (
+  defaults: Partial<ElevationProps> = {}
+) => ({
+  elevation: {
+    type: Number,
+    default: defaults.elevation,
+  },
+})
+
+// Effect
+export function useElevation ({ elevation }: ElevationProps) {
+  const classes: Dictionary<boolean> = {}
+
+  if (
+    elevation != null &&
+    !isNaN(parseInt(elevation))
+  ) {
+    classes[`elevation-${elevation}`] = true
+  }
+
+  const elevationClasses = computed(() => classes)
```

Comment:
why not just 
```ts
const elevationClasses = computed(() => {
  if (props.elevation == null || isNan(parseInt(props.elevation))) return {}

  return { [`elevation-${props.elevation}`]: true }
})
```

Reply:
Additionaly there could be a warning (emitted in watcher) if elevation is `NaN` or `< 0` or `> MAX_ELEVATION`, maybe only in dev mode

Reply:
Refactored the function. I like the assertion, we should push that to a post 3 release though.